sdk/core/Azure.Core.Experimental/src/DynamicRequest.cs

```diff
@@ -0,0 +1,131 @@
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents an HTTP request with <see cref="DynamicJson"/> content.
+    /// </summary>
+    public class DynamicRequest : Request
+    {
+        private Request Request { get; }
+        private HttpPipeline HttpPipeline { get; }
+
+        private static readonly Encoding Utf8NoBom = new UTF8Encoding(false, true);
+
+        /// <inheritdoc />
+        public override RequestContent? Content
+        {
+            get => DynamicContent.Create(Body);
+
+            set {
+                MemoryStream ms = new MemoryStream();
+                if (value != null)
+                {
+                    value.WriteTo(ms, default);
+                    ms.Seek(0, SeekOrigin.Begin);
+                }
+                using (StreamReader sr = new StreamReader(ms, Utf8NoBom))
+                {
+                    Body = new DynamicJson(sr.ReadToEnd());
+                }
+                Request.Content = value;
+            }
+        }
+
+        // TODO(matell): How does the initialization here play into the ability to send a request with an "empty" body?
+        /// <summary>
+        /// The JSON body of request.
+        /// </summary>
+        public DynamicJson Body { get; set; } = DynamicJson.Object();
+
+        // TODO(matell): In Krzysztof's prototype we also took DiagnosticScope as a parameter, do we still need that?
+        /// <summary>
+        /// Creates an instance of <see cref="RequestContent"/> that wraps <see cref="DynamicJson"/> content.
+        /// </summary>
+        /// <param name="request">The <see cref="Request"/> to send.</param>
+        /// <param name="pipeline">The HTTP pipeline for sending and receiving REST requests and responses.</param>
+        public DynamicRequest(Request request, HttpPipeline pipeline)
+        {
+            Request = request;
+            HttpPipeline = pipeline;
+        }
+
+        /// <summary>
+        /// Send the request asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The response dynamically typed in a <see cref="DynamicResponse"/>.</returns>
+        public async Task<DynamicResponse> SendAsync(CancellationToken cancellationToken = default)
+        {
+            // Since we are sending the underlying request, we need to copy the Content on to it, or we'll lose the body.
+            Request.Content = Content;
+
+            Response res = await HttpPipeline.SendRequestAsync(Request, cancellationToken).ConfigureAwait(false);
+            DynamicJson dynamicContent;
+
+            if (res.ContentStream != null)
+            {
+                using (StreamReader sr = new StreamReader(res.ContentStream, encoding: Utf8NoBom, detectEncodingFromByteOrderMarks:true, bufferSize: 1024, leaveOpen: true))
+                {
+                    dynamicContent = new DynamicJson(await sr.ReadToEndAsync().ConfigureAwait(false));
```

Comment:
I think you should use the JsonElement ctor to initialize this instance.  The JE ctor avoids (or should avoid) bunch of costs related to string allocations and transcoding, i.e. JsonDocument can read UTF8 streams directly.